src/python/pants/core/goals/tailor.py

```diff
@@ -261,6 +261,13 @@ def merge(cls, tgts_iters: Iterable[PutativeTargets]) -> PutativeTargets:
         return cls(all_tgts)
 
 
+@rule(polymorphic=True)
+async def get_putative_targets(
```

Comment:
Verb-wise: I would prefer `generate_putative_targets` or `build_putative_targets` -- `get` seems too generic to me.

Reply:
Or maybe `resolve_putative_targets`